setup.py

```diff
@@ -1,19 +1,134 @@
-from setuptools import setup, find_packages
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
 
-VERSION = '0.1'
+import os
+import sys
 
+from setuptools import setup
+from setuptools.command.install import install
 
-# NOTE: Use Pipfile & Pipfile.lock to manage dependencies\
-INSTALL_REQUIRES = []
-TESTS_REQUIRE = []
 
-setup(name='nucypher',
-      version=VERSION,
-      description='A proxy re-encryption network to empower privacy in decentralized systems.',
+#
+# Metadata
+#
+
+PACKAGE_NAME = 'nucypher'
+BASE_DIR = os.path.dirname(__file__)
+
+ABOUT = dict()
+with open(os.path.join(BASE_DIR, PACKAGE_NAME, "__about__.py")) as f:
+    exec(f.read(), ABOUT)
+
+
+with open(os.path.join(BASE_DIR, "README.md")) as f:
+    long_description = f.read()
+
+
+#
+# Utilities
+#
+
+class VerifyVersionCommand(install):
+    """Custom command to verify that the git tag matches our version"""
+    description = 'verify that the git tag matches our version'
+
+    def run(self):
+        tag = os.getenv('CIRCLE_TAG')
+        if tag.startswith('v'):
+            tag = tag[1:]
+
+        version = ABOUT['__version__']
+        if version.startswith('v'):
+            version = version[1:]
+
+        if tag != version:
+            info = "Git tag: {0} does not match the version of this app: {1}".format(
+                os.getenv('CIRCLE_TAG'), ABOUT['__version__']
+            )
+            sys.exit(info)
+
+
+#
+#  Dependencies
+#
+
+INSTALL_REQUIRES = [
+
+    # NuCypher
+    'umbral',
+    'constant-sorrow',
+    'bytestringSplitter',
+    'hendrix>=3.1.0',
+
+    # Third Party (General)
+    'cryptography>=2.3',
+    'pysha3',
+    'requests',
+    'sqlalchemy',
+    'apistar==0.5.42',
+    'tzlocal==2.0.0b1',
```

Comment:
This doesn't work for me. As soon as `tzlocal==2.0.0b1` got into `setup.py`, `nucypher` command fails for me. It shouldn't be pinned to this version neither in `setup.py`, nor in `Pipfile`

Reply:
@KPrasch and I discussed this, taking care of this now is going to needlessly break some dev environments. We're going to push this for a later PR.

Reply:
Sure. I guess, we should have an issue for this, or add a comment in an open PR where this is to be done, in order to not forget.
Although, it wouldn't even install for me without me fixing it every time, so I won't forget :joy: